src/Raven.Server/Documents/Sharding/Handlers/Processors/Revisions/ShardedRevisionsHandlerProcessorForGetRevisions.cs

```diff
@@ -0,0 +1,101 @@
+    {
+        public ShardedRevisionsHandlerProcessorForGetRevisions([NotNull] ShardedDatabaseRequestHandler requestHandler) : base(requestHandler, requestHandler.ContextPool)
+        {
+        }
+
+        protected override async ValueTask GetRevisionByChangeVectorAsync(Microsoft.Extensions.Primitives.StringValues changeVectors, bool metadataOnly, CancellationToken token)
+        {
+            using (ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            {
+                var cmd = new ShardedGetRevisionsByChangeVectorsOperation(changeVectors.ToArray(), metadataOnly, context);
+                
+                var res = await RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(cmd, token);
+
+                if (res == null && changeVectors.Count == 1)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                    return;
+                }
+
+                var blittable = RequestHandler.GetBoolValueQueryString("blittable", required: false) ?? false;
+                
+                if (blittable)
+                {
+                    WriteRevisionsBlittable(context, res, out long numberOfResults, out long totalDocumentsSizeInBytes);
+                }
+                else
+                {
+                    await WriteRevisionsResult(context, res);
+                }
+            }
+        }
+
+        protected override async ValueTask GetRevisionsAsync(bool metadataOnly, CancellationToken token)
+        {
+            var id = RequestHandler.GetQueryStringValueAndAssertIfSingleAndNotEmpty("id");
+            var before = RequestHandler.GetDateTimeQueryString("before", required: false);
+            var start = RequestHandler.GetStart();
+            var pageSize = RequestHandler.GetPageSize();
+
+            GetRawRevisionsCommand cmd;
+            if (before.HasValue)
+            {
+                cmd = new GetRawRevisionsCommand(RequestHandler, id, before.Value);
+            }
+            else
+            {
+                cmd = new GetRawRevisionsCommand(RequestHandler, id, start, pageSize, metadataOnly);
+            }
+
+            int shardIndex;
+            using (RequestHandler.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                shardIndex = RequestHandler.DatabaseContext.GetShardNumber(context, id);
+            }
+            
+            //cmd writes the response to stream as is
+            await RequestHandler.ShardExecutor.ExecuteSingleShardAsync(cmd, shardIndex, token);
+        }
+
+        protected override void CheckNotModified(string actualEtag)
+        {
+            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Minor, "Implement AddPagingPerformanceHint");
```

Comment:
wrong message and please put Normal severity

Reply:
done